include/exadg/operators/operator_base.cpp

```diff
@@ -306,6 +360,68 @@ OperatorBase<dim, Number, n_components>::apply_add(VectorType & dst, VectorType
   }
 }
 
+template<int dim, typename Number, int n_components>
+void
+OperatorBase<dim, Number, n_components>::apply_matrix_based(VectorType &       dst,
+                                                            VectorType const & src) const
+{
+  if(this->data.sparse_matrix_type == SparseMatrixType::Trilinos)
+  {
+#ifdef DEAL_II_WITH_TRILINOS
+    if constexpr(std::is_same_v<Number, double>)
+    {
+      system_matrix_trilinos.vmult(dst, src);
+    }
+    else
+    {
+      // create temporal vectors of type double
+      dealii::LinearAlgebra::distributed::Vector<double> dst_double, src_double;
+      dst_double.reinit(dst, false);
```

Comment:
I adapted this code from another place in ExaDG. My question is whether we really need `false` here as the second argument, because `dst` is overwritten anyway in `vmult()`, isn't it?

Reply:
Do you mean places like https://github.com/exadg/exadg/blob/c62fd0a241beab24a91cea6fd64680cbfb2c17cd/include/exadg/operators/operator_base.cpp#L319-L328 Indeed, we should be able to replace it by `true` here (where true means "omit initializing the vector entries to zero"). The reason why the other function needs it is because that other place runs a matrix-free loop that adds into the vector, so we must start from zero. (Or we could ask the loop to set it to zero, it does not matter too much for that particular function.)

Reply:
I probably changed this code so that the above reference might have disappeared. Sorry.

I meant the `reinit()` before we do the `vmult()`. So the question would be whether we need a `dst_double=0` prior to calling `vmult()`. I would argue no because `vmult()` overwrites the values anyway. But I am asking nevertheless because we also had in the past special aspects e.g. related to ghost values where I am not completely sure whether such a change might invalidate something.

Reply:
I agree, for matrix-based `vmult` one should not need a `operator=(0.0)` because the matrix-based solver will overwrite the content in their implementations (as far as I know). The matrix-free code is special in the sense of ghost entries, where we had bugs in the past (hopefully fixed today), but we don't expose the ghost range to PETSc/Trilinos, so I think this should be fine.